packages/fluentui/react-bindings/src/hooks/useTrigger.tsx

```diff
@@ -0,0 +1,86 @@
+import { Ref } from '@fluentui/react-component-ref';
+import * as React from 'react';
+import * as ReactIs from 'react-is';
+
+import { childrenExist } from '../utils/childrenExist';
+
+type UseTriggerOptions = {
+  children?: React.ReactNode;
+  trigger?: React.ReactElement;
+};
+
+// https://github.com/facebook/react/blob/c4e0768d7487a9359b74986e3b07841d2520f593/packages/react-dom/src/events/getListener.js#L15-L22
+function isDisabledInteractive(node: Node): boolean {
+  return (
+    (node.nodeName === 'BUTTON' ||
+      node.nodeName === 'INPUT' ||
+      node.nodeName === 'SELECT' ||
+      node.nodeName === 'TEXTAREA') &&
+    (node as HTMLButtonElement).disabled
+  );
+}
+
+function isInteractiveFilter(node: Node) {
+  return isDisabledInteractive(node) ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_SKIP;
+}
+
+export function useTrigger(props: UseTriggerOptions): React.ReactElement | null {
+  const trigger = childrenExist(props.children) ? props.children : props.trigger;
+  const element = trigger ? React.Children.only(trigger) : null;
+
+  if (element !== null) {
+    if (!React.isValidElement(element)) {
+      throw new Error('useTrigger(): An invalid value was passed, please pass a valid React element as a trigger');
+    }
+
+    if (ReactIs.isFragment(element)) {
+      throw new Error(
+        'useTrigger(): A "React.Fragment" cannot be used as a "trigger" as it will be impossible to spread props on it',
+      );
+    }
+  }
```

Comment:
Could consider wrapping this if block in a dev mode check, we probably don't need to throw in production builds.

Reply:
Moved 👍